crates/controller/src/hud_components.rs

```diff
@@ -0,0 +1,52 @@
+use bevy::prelude::*;
+use crate::mouse::HudVisibleNode;
+
+const HUD_COLOR: Color = Color::BLACK;
+
+pub(crate) fn spawn_details(
+    commands: &mut ChildBuilder,
+) {
+    commands.spawn((side_node_bundle(HUD_COLOR), HudVisibleNode));
+}
+
+pub(crate) fn spawn_action_bar(
+    commands: &mut ChildBuilder,
+) {
+    commands.spawn((NodeBundle {
+        style: Style {
+            size: Size { width: Val::Percent(60.), height: Val::Percent(15.) },
+            flex_direction: FlexDirection::Row,
+            flex_grow: 1.0,
+            ..simple_style()
+        },
+        background_color: HUD_COLOR.into(),
+        ..default()
+    }, HudVisibleNode));
+}
+
+pub(crate) fn spawn_map(
+    commands: &mut ChildBuilder,
+) {
+    commands.spawn((side_node_bundle(HUD_COLOR), HudVisibleNode));
+}
+
+fn side_node_bundle(color: Color) -> NodeBundle {
+    NodeBundle {
+        style: Style {
+            size: Size { width: Val::Percent(20.), height: Val::Percent(30.) },
+            align_items: AlignItems::Stretch,
```

Comment:
Why are non-default values needed? At a first glance, it seems like only `size` is relevant in current state.